Userland/Libraries/LibC/stdio.cpp

```diff
@@ -1287,19 +1325,27 @@ int vscanf(const char* fmt, va_list ap)
     return vfscanf(stdin, fmt, ap);
 }
 
-void flockfile([[maybe_unused]] FILE* filehandle)
+void flockfile(FILE* filehandle)
+{
+    VERIFY(filehandle);
+    ScopedFileLock::lock(filehandle);
+}
+
+int ftrylockfile(FILE* filehandle)
 {
-    dbgln("FIXME: Implement flockfile()");
+    VERIFY(filehandle);
+    return ScopedFileLock::try_lock(filehandle);
```

Comment:
This is bogus. `ScopedFileLock` will unlock the file as soon as it is out of scope. This does not actually lock the file in anyway.

Reply:
I added static functions to `ScopedFileLock` that only lock or unlock the file. The name might be misleading, but we should keep the lock functions of `FILE` private and not add any friend functions.

Reply:
Then please change that. `ScopedFileLock` locks a file within a scope. Please don't add things that allow it to do things outside a scope.